### PR TITLE
[RAC] fix: unexpected TableRow children render call

### PR DESCRIPTION
**Pull request by ryo-manba**
Closes https://github.com/adobe/react-spectrum/issues/5241

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Checked locally using Storybook. Verified the Row component's children render function no longer receives unexpected arguments.

### CODE REVIEW

File: packages/react-aria-components/src/Table.tsx [hunk 1]
```diff
@@ -1040,8 +1040,9 @@ function TableRow<T>({item}: {item: GridNode<T>}) {
 
   let props = item.props as RowProps<unknown>;
   let isDragging = dragState && dragState.isDragging(item.key);
+  let {children:_, ...restProps} = props;
```
Comment (snowystinger) [inline packages/react-aria-components/src/Table.tsx:1043]:
```suggestion
  // eslint-disable-next-line no-unused-vars
  let {children:_, ...restProps} = props;
```

File: packages/react-aria-components/src/Table.tsx [hunk 2]
```diff
@@ -1040,8 +1040,10 @@ function TableRow<T>({item}: {item: GridNode<T>}) {
 
   let props = item.props as RowProps<unknown>;
   let isDragging = dragState && dragState.isDragging(item.key);
+  // eslint-disable-next-line no-unused-vars
+  let {children:_, ...restProps} = props;
```
Comment (snowystinger) [inline packages/react-aria-components/src/Table.tsx:1044]:
```suggestion
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  let {children: _, ...restProps} = props;
```

**Review by snowystinger — COMMENTED** [review]
Description says checked locally to test. I don't see any stories in our storybook which would allow for checking this. Could you supply a story or even better, a test?
https://react-spectrum.adobe.com/contribute.html#pull-requests

**Comment by ryo-manba** [comment]
@snowystinger 
Thank you for the feedback.

I've added a new story, TableDynamicExample, to demonstrate the behavior. This story displays a table with dynamic rows and columns, allowing us to verify the Row component's children rendering.

File: packages/react-aria-components/stories/index.stories.tsx [hunk 3]
```diff
@@ -863,6 +863,40 @@ export const TableExample = () => {
...
+        {(column) => (
+          <Column isRowHeader={column.isRowHeader}>{column.name}</Column>
+        )}
+      </TableHeader>
+      <TableBody items={rows}>
+        {(item) => (
+          <Row columns={columns}>
+            {(column) => {
+              return <Cell>{item[column.key]}</Cell>;
+            }}
```
Comment (ryo-manba) [inline packages/react-aria-components/stories/index.stories.tsx:892]: To better understand the column data, consider logging it:

```ts
{(column) => {
  console.log(column);
  return <Cell>{item[column.key]}</Cell>;
}}
```

**Comment by snowystinger** [comment]
GET_BUILD

**Comment by snowystinger** [comment]
GET_BUILD

**Review by reidbarber — APPROVED** [review]
LGTM

File: packages/react-aria-components/test/Table.test.js [hunk 4]
```diff
@@ -610,6 +610,49 @@ describe('Table', () => {
...
+
+    // React canary only calls render function once, vs twice in React 18, 17 and 16.
+    // Every call should be the same, so just loop over them.
+    expect(renderRow.mock.calls.length).toBeGreaterThanOrEqual(1);
+    renderRow.mock.calls.forEach((call) => {
+      expect(call[0]).toBe(columns[0]);
+    });
+    renderRow.mockReset();
+
+    // We do not currently call the renderProps function on any of these changes
```
Comment (snowystinger) [inline packages/react-aria-components/test/Table.test.js:649]: I've opened an issue here https://github.com/adobe/react-spectrum/issues/5299

**Review by snowystinger — APPROVED** [review]
Thanks for the PR

**Pull request merged**